Guide/routing.markdown

```diff
@@ -159,9 +159,68 @@ instance HasPath PostsController where
     pathTo ShowAllMyPostsAction = "/posts"
 ```
 
+### Beautiful URLs
+
+Let's say we want to give our blog post application a beautiful url structures for SEO reasons. Our controller is defined as:
+
+```haskell
+data PostsController
+    = ShowPostAction { postId :: !(Id Post) }
+```
+
+We want our urls to look like this:
+
+```html
+/posts/an-example-blog-post
+```
+
+Additionally we also want to accept permalinks with the id like this:
+
+```
+/posts/f85dc0bc-fc11-4341-a4e3-e047074a7982
+```
+
+To accept urls like this, we first need to make some changes to our data structure. We have to make the `postId` optional. Additonally we need to have a parameter for the url slug:
+
+```haskell
+data PostsController
+    = ShowPostAction { postId :: !(Maybe (Id Post)), slug :: !(Maybe Text) }
+```
+
+This will also require us to make changes to our action implementation:
+
+```haskell
+action ShowPostAction { postId, slug } = do 
+    post <- case slug of
+            Just slug -> query @Post |> filterWhere (#slug, slug) |> fetchOne
+            Nothing   -> fetchOne postId
```

Comment:
Does `fetchOne` accept a `Maybe (Id Post)`?

Reply:
Yes :) https://github.com/digitallyinduced/ihp/blob/master/IHP/QueryBuilder.hs#L430 It's a super useful helper instance because this pattern comes up a lot in practise.